test/lit/passes/optimize-instructions-exact.wast

```diff
@@ -0,0 +1,66 @@
+;; NOTE: Assertions have been generated by update_lit_checks.py and should not be edited.
+
+;; RUN: wasm-opt %s -all --disable-custom-descriptors --optimize-instructions -S -o - | filecheck %s
+
+(module
+ ;; CHECK:      (type $super (sub (struct)))
+ (type $super (sub (struct)))
+
+ ;; CHECK:      (func $ref-cast-exact-fallthrough (type $1) (param $0 (ref (exact $super))) (result (ref $super))
+ ;; CHECK-NEXT:  (local $1 (ref $super))
+ ;; CHECK-NEXT:  (local $2 (ref (exact $super)))
+ ;; CHECK-NEXT:  (drop
+ ;; CHECK-NEXT:   (local.tee $1
+ ;; CHECK-NEXT:    (local.tee $2
+ ;; CHECK-NEXT:     (local.get $0)
+ ;; CHECK-NEXT:    )
+ ;; CHECK-NEXT:   )
+ ;; CHECK-NEXT:  )
+ ;; CHECK-NEXT:  (local.get $2)
+ ;; CHECK-NEXT: )
+ (func $ref-cast-exact-fallthrough (param (ref (exact $super))) (result (ref $super))
+  (local (ref $super))
+  ;; We should find that the local.get is the most precise fallthrough value and
+  ;; hoist it to eliminate the cast.
+  (ref.cast (ref $super)
+   (local.tee 1
+    (local.get 0)
```

Comment:
Might be clearer to use var names here. And if the param name were `$exact` (to contrast with the local `$inexact` or just `$struct`) then readability would be even better imo